// emon/esm2022/lib/basic/input/text/text.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TextComponent } from "./text.component";
import { FormsModule, ReactiveFormsModule } from "@angular/forms";
import { IconModule } from "../../icon/icon.module";
import { BrowserModule } from "@angular/platform-browser";
import * as i0 from "@angular/core";
export class TextModule {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.9", ngImport: i0, type: TextModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "18.2.9", ngImport: i0, type: TextModule, declarations: [TextComponent], imports: [BrowserModule,
            CommonModule,
            FormsModule,
            IconModule,
            ReactiveFormsModule], exports: [TextComponent] });
    static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "18.2.9", ngImport: i0, type: TextModule, imports: [BrowserModule,
            CommonModule,
            FormsModule,
            IconModule,
            ReactiveFormsModule] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.9", ngImport: i0, type: TextModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [TextComponent],
                    imports: [
                        BrowserModule,
                        CommonModule,
                        FormsModule,
                        IconModule,
                        ReactiveFormsModule
                    ],
                    exports: [TextComponent]
                }]
        }] });
