// emon/esm2022/lib/basic/input/number/number.module.mjs
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { FormsModule, ReactiveFormsModule } from "@angular/forms";
import { IconModule } from "../../icon/icon.module";
import { NumberComponent } from "./number.component";
import { TextModule } from "../text/text.module";
import * as i0 from "@angular/core";
export class NumberModule {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.9", ngImport: i0, type: NumberModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "18.2.9", ngImport: i0, type: NumberModule, declarations: [NumberComponent], imports: [CommonModule,
            FormsModule,
            IconModule,
            ReactiveFormsModule,
            TextModule], exports: [NumberComponent] });
    static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "18.2.9", ngImport: i0, type: NumberModule, imports: [CommonModule,
            FormsModule,
            IconModule,
            ReactiveFormsModule,
            TextModule] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.9", ngImport: i0, type: NumberModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NumberComponent],
                    imports: [
                        CommonModule,
                        FormsModule,
                        IconModule,
                        ReactiveFormsModule,
                        TextModule
                    ],
                    exports: [
                        NumberComponent
                    ]
                }]
        }] });
