// emon/esm2022/lib/core/input-base/input-base.mjs
import { Component, Input } from "@angular/core";
import { BaseControlValueAccessor } from "../control-value-accessor/base-cva.class";
import * as i0 from "@angular/core";
export class InputBaseComponent extends BaseControlValueAccessor {
    styleClass;
    validationStatus;
    label;
    isDisabled;
    pattern = '';
    width = '200px';
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.9", ngImport: i0, type: InputBaseComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "18.2.9", type: InputBaseComponent, isStandalone: true, selector: "ng-component", inputs: { styleClass: "styleClass", validationStatus: "validationStatus", label: "label", isDisabled: "isDisabled", pattern: "pattern", width: "width" }, usesInheritance: true, ngImport: i0, template: '', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.9", ngImport: i0, type: InputBaseComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '',
                    standalone: true,
                    imports: [InputBaseComponent],
                    template: ''
                }]
        }], propDecorators: { styleClass: [{
                type: Input
            }], validationStatus: [{
                type: Input
            }], label: [{
                type: Input
            }], isDisabled: [{
                type: Input
            }], pattern: [{
                type: Input
            }], width: [{
                type: Input
            }] } });
