// emon/esm2022/lib/basic/input/checkbox/checkbox.module.mjs
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { CheckboxComponent } from "./checkbox.component";
import { FormsModule, ReactiveFormsModule } from "@angular/forms";
import { IconModule } from "../../icon/icon.module";
import * as i0 from "@angular/core";
export class CheckboxModule {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.9", ngImport: i0, type: CheckboxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "18.2.9", ngImport: i0, type: CheckboxModule, declarations: [CheckboxComponent], imports: [CommonModule,
            FormsModule,
            IconModule,
            ReactiveFormsModule], exports: [CheckboxComponent] });
    static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "18.2.9", ngImport: i0, type: CheckboxModule, imports: [CommonModule,
            FormsModule,
            IconModule,
            ReactiveFormsModule] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.9", ngImport: i0, type: CheckboxModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CheckboxComponent],
                    imports: [
                        CommonModule,
                        FormsModule,
                        IconModule,
                        ReactiveFormsModule
                    ],
                    exports: [
                        CheckboxComponent
                    ]
                }]
        }] });
